gym/envs/mujoco/ant_v4.py

```diff
@@ -0,0 +1,298 @@
+
+    ### Starting State
+    All observations start in state
+    (0.0, 0.0,  0.75, 1.0, 0.0  ... 0.0) with a uniform noise in the range
+    of [-0.1, 0.1] added to the positional values and standard normal noise
+    with 0 mean and 0.1 standard deviation added to the velocity values for
+    stochasticity. Note that the initial z coordinate is intentionally selected
+    to be slightly high, thereby indicating a standing up ant. The initial orientation
+    is designed to make it face forward as well.
+
+    ### Episode Termination
+    The episode terminates when any of the following happens:
+
+    1. The episode duration reaches a 1000 timesteps
+    2. Any of the state space values is no longer finite
+    3. The y-orientation (index 2) in the state is **not** in the range `[0.2, 1.0]`
+
+    ### Arguments
+
+    No additional arguments are currently supported (in v2 and lower), but modifications
+    can be made to the XML file in the assets folder (or by changing the path to a modified
+    XML file in another folder).
+
+    ```
+    env = gym.make('Ant-v2')
+    ```
+
+    v3 and beyond take gym.make kwargs such as xml_file, ctrl_cost_weight, reset_noise_scale etc.
+
+    ```
+    env = gym.make('Ant-v3', ctrl_cost_weight=0.1, ...)
+    ```
+
+    ### Version History
+    * v4: all mujoco environments now use the mujoco binidings in mujoco>=2.1.3
+    * v3: support for gym.make kwargs such as xml_file, ctrl_cost_weight, reset_noise_scale etc. rgb rendering comes from tracking camera (so agent does not run away from screen)
+    * v2: All continuous control environments now use mujoco_py >= 1.50
+    * v1: max_time_steps raised to 1000 for robot based tasks. Added reward_threshold to environments.
+    * v0: Initial versions release (1.0.0)
+    """
+
+    def __init__(
+        self,
+        xml_file="ant.xml",
+        ctrl_cost_weight=0.5,
+        contact_cost_weight=5e-4,
+        healthy_reward=1.0,
+        terminate_when_unhealthy=True,
+        healthy_z_range=(0.2, 1.0),
+        contact_force_range=(-1.0, 1.0),
+        reset_noise_scale=0.1,
+        exclude_current_positions_from_observation=True,
+    ):
+        utils.EzPickle.__init__(**locals())
+
+        self._ctrl_cost_weight = ctrl_cost_weight
+        self._contact_cost_weight = contact_cost_weight
+
+        self._healthy_reward = healthy_reward
+        self._terminate_when_unhealthy = terminate_when_unhealthy
+        self._healthy_z_range = healthy_z_range
+
+        self._contact_force_range = contact_force_range
+
+        self._reset_noise_scale = reset_noise_scale
+
+        self._exclude_current_positions_from_observation = (
+            exclude_current_positions_from_observation
+        )
+
+        mujoco_env.MujocoEnv.__init__(self, xml_file, 5)
+
+    @property
+    def healthy_reward(self):
+        return (
+            float(self.is_healthy or self._terminate_when_unhealthy)
+            * self._healthy_reward
+        )
+
```

Comment:
I think this has a bug:

- when `terminate_when_unhealthy` is true (default setting), `self.healthy_reward` should be conditioned on `is_healthy`; however, the above code is not (always `self._healthy_reward`)
- when `terminate_when_unhealthy` is false, `self.healthy_reward` should always be `self._healthy_reward` (done is always False); however, the above code is not (conditioned on `self.is_healthy`)

```suggestion
    @property
    def healthy_reward(self):
        return (
            float(self.is_healthy or not self._terminate_when_unhealthy)
            * self._healthy_reward
        )

```